assets/src/BlockEditorValidation.js

```diff
@@ -0,0 +1,93 @@
+const { registerPlugin } = wp.plugins;
+const { PluginPrePublishPanel } = wp.editPost;
+const { dispatch } = wp.data;
+const {__} = wp.i18n;
+
+export const blockEditorValidation = () => {
+  registerPlugin( 'pre-publish-checklist', { render: PrePublishCheckList } );
+};
+
+const PrePublishCheckList = () => {
+  let lockPost = false;
+
+  let checkListMsg = '';
+  let classname    = '';
+
+  // "Analytics & Tracking" fields validation.
+  let $htmlbody   = $( 'html, body' );
+  // Filter CMB2 fields that have the 'data-validation' attribute set to 'required'.
+  let $toValidate = $( '[data-validation]' );
+
+  let analyticsErrorFlag   = false;
```

Comment:
This should probably be something more generic than `analyticsErrorFlag` as other fields could be validated too?